Add Ability to Query Remote Collins

michaeljs1990 opened the pull request:
This adds in the remoteLookup flag as seen in the asset API on
https://tumblr.github.io/collins/api.html. both r and R were taken
so I was just out of luck on that and Z came to mind but open to
other ideas. Possibly this is always the default behavior as well
and a config option should be added so you don't have to remember to
query all the collins.

Integration Test:
```
mschuett-C02TC2VYGTDY:collins-cli schuettm$ bundle exec collins find -Z -t U002144
U002144	pop-cba8fa2a.totallyarealdomain.net	popnode	Provisioned	PRODUCTION	ADMIN	POPNODE	
mschuett-C02TC2VYGTDY:collins-cli schuettm$ bundle exec collins find -t U002144
No assets found
```

michaeljs1990 committed:
Add Ability to Query Remote Collins

This adds in the remoteLookup flag as seen in the asset API on
https://tumblr.github.io/collins/api.html. both r and R were taken
so I was just out of luck on that and Z came to mind but open to
other ideas. Possibly this is always the default behavior as well
and a config option should be added so you don't have to remember to
query all the collins.

byxorna commented on lib/collins/cli/find.rb:
```diff
@@ -40,6 +40,7 @@ def parse!(argv = ARGV)
         opts.banner = "Usage: #{PROG_NAME} [options] [hostnamepattern]"
         opts.separator "Query options:"
         opts.on('-t','--tag TAG[,...]',Array, "Assets with tag[s] TAG") {|v| search_attrs[:tag] = v}
+        opts.on('-Z','--remote',"Query remote datacenters for asset") {|v| search_attrs[:remoteLookup] = v}
```
can you set `search_attrs[:remoteLookup] = false` as default, or are you comfortable relying on the API's defaults?

Also, maybe make the long flag map to the attribute in the API? `--remote-lookup`?

byxorna reviewed (approved):
This looks awesome! Comments inline

michaeljs1990 commented:
Switched out the long flag to be remote-lookup and added in a default since this flag wouldn't make sense if tumblr/collins switched remote lookup to be true by default.

byxorna commented:
Thanks @michaeljs1990 ! ill cut a release in a sec

michaeljs1990 commented:
Awesome! Thanks for getting this merged in.

byxorna commented:
@michaeljs1990 heres the jawn: https://rubygems.org/gems/collins-cli/versions/0.2.12
